Re: #3037 Rename sample_ppc to sample_posterior_predictive

Dpananos opened the pull request:
As per #3037:

- Rename `sample` to `sample_posterior`.  Keep `sample` as an alias and add deprecation warning.
- Rename `sample_ppc` to `sample_posterior_predictive`.  Keep `sample_ppc` as an alias and add deprecation warning.
- Rename `sample_ppc_w` to `sample_posterior_predictive_w`.  Keep `sample_ppc_w` as an alias and add deprecation warning.

Minor formatting to adhere to pep8.

Dpananos committed:
sampling edits

Renamed sample, sample_ppc, and sample_ppc_w.  Added deprecation warnings.

Dpananos committed:
Removed .coverage

Dpananos committed:
Rename/alias

Dpananos committed:
Adjust white space

twiecki commented on pymc3/sampling.py:
```diff
@@ -481,6 +490,11 @@ def sample(draws=500, step=None, init='auto', n_init=200000, start=None, trace=N
     return trace
 
+def sample(*args,**kwargs):
+    """This method is deprecated.  Please use :func:`~sampling.sample_posterior`"""
```
Actually I don't think we want to deprecate this function, just because it's the main entry point for every user. Instead, I would propose to make `sample_posterior` an alias to `sample()`. CC @junpenglao @fonnesbeck

junpenglao replied:
Agree.

Dpananos replied:
Ah, sorry about that.  Will change.

twiecki commented:
@Dpananos Thanks! I'm afraid we also need to change all docs and examples referring to `sample_ppc`.

Dpananos committed:
Edits to Docs

Swapped sample_ppc and sample_ppc_w for sample_posterior_predictive and sample_posterior_predictive_w

Dpananos commented:
Docs edited.

fonnesbeck commented:
Is this ready to go?

Dpananos commented:
Last i checked, the functions were renamed, aliased, and docs were changed. I’ll double check today after work and give the thumbs up.

Demetri Pananos
Ph.D Candidate
Department of Epidemiology and Biostatistics
Schulich School of Medicine and Dentistry
Western University
On Jul 25, 2018, 11:57 AM -0400, Chris Fonnesbeck <notifications@github.com>, wrote:
> Is this ready to go?
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

Dpananos committed:
pep8 Edits

Dpananos commented:
Looks good on my end.  Don't know why the .ipynb files seem to be conflicting.

fonnesbeck commented:
@Dpananos try merging master into your branch, and then pushing again.

Dpananos committed:
Merge remote-tracking branch 'upstream/master' into sampling-edits

Dpananos committed:
Merge branch 'master' into sampling-edits

Dpananos committed:
Revert "Merge remote-tracking branch 'upstream/master' into sampling-edits"

This reverts commit b64bc12db38610d81825395fa9ac66e94618e605.

Dpananos commented:
Yea, I don't know what has happened.  Not the greatest with the command line tools.  I can just nuke the branch, refork master, and start again.  Changes are not that extensive.  Unless someone can walk me through steps to undo the commit and merge master into my branch.

Dpananos committed:
s
